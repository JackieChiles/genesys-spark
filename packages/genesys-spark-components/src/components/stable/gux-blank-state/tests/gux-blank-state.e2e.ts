import { a11yCheck, newSparkE2EPage } from '../../../../test/e2eTestUtils';

describe('gux-blank-state', () => {
  it('renders', async () => {
    const html = `
      <gux-blank-state lang="en">
        <gux-icon slot="image" icon-name="bot" decorative="true"></gux-icon>
        <div slot="primary-message">Sorry, something went wrong.</div>
        <div slot="additional-guidance">Please refresh this page to try again.</div>
        <button slot="call-to-action" type="button" onclick="notify(event)">
          Call to action
        </button>
      </gux-blank-state>`;
    const page = await newSparkE2EPage({ html });
    const element = await page.find('gux-blank-state');

    await a11yCheck(page);

    expect(element).toHaveAttribute('hydrated');
    expect(element.outerHTML).toMatchSnapshot();
  });
});
