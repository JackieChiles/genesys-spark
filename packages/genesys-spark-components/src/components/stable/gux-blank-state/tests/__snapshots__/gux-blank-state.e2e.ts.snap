// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-blank-state renders 1`] = `"<gux-blank-state lang="en" hydrated=""><template shadowrootmode="open"><div class="gux-container"><div class="gux-image"><slot name="image"></slot></div><div class="gux-message"><slot name="primary-message"></slot></div><div class="gux-guidance"><slot name="additional-guidance"></slot></div><gux-button-slot accent="primary" hydrated=""><slot name="call-to-action"></slot></gux-button-slot></div></template> <gux-icon slot="image" icon-name="bot" decorative="true" size="inherit" hydrated=""></gux-icon> <div slot="primary-message">Sorry, something went wrong.</div> <div slot="additional-guidance">Please refresh this page to try again.</div> <button slot="call-to-action" type="button" onclick="notify(event)"> Call to action </button> </gux-blank-state>"`;
